wpas-hl7-translator/src/main/java/wales/nhs/dhcw/inthub/wpasHl7/Main.java

```diff
@@ -39,11 +39,12 @@ public static void main(String[] args) throws JAXBException {
             while (APP_RUNNING) {
                 receiverClient.receiveMessages(MAX_BATCH_SIZE, message -> {
                     var messageBody = message.getBody();
+                    var enqueuedTime = message.getEnqueuedTime();
                     LOGGER.debug("Received message: {}", messageBody);
 
                     try {
                         var mainData = parser.parse(messageBody.toStream());
-                        var hl7Data = translator.translate(mainData);
+                        var hl7Data = translator.translate(mainData,enqueuedTime);
```

Comment:
we could wrap both main data and addtional properties in a class to support future extensions